tests/test_train.py

```diff
@@ -45,32 +46,41 @@ def test_package_model(trained_rasa_model, parameters):
     assert file_name.endswith(".tar.gz")
 
 
-@pytest.fixture
-def move_tempdir():
-    # Create a new *empty* tmp directory
-    shutil.rmtree(TEST_TEMP, ignore_errors=True)
-    os.mkdir(TEST_TEMP)
-    tempfile.tempdir = TEST_TEMP
-    yield
-    tempfile.tempdir = None
-    shutil.rmtree(TEST_TEMP)
+def count_temp_rasa_files(directory: Text) -> int:
+    return len(
+        [
+            entry
+            for entry in os.listdir(directory)
+            if not any(
+                [
+                    # Ignore the following files/directories:
+                    entry == "__pycache__",  # Python bytecode
+                    entry.endswith(".py")  # Temp .py files created by TF
+                    # Anything else is considered to be created by Rasa
+                ]
+            )
+        ]
+    )
 
 
 def test_train_temp_files(
-    move_tempdir,
-    default_domain_path,
-    default_stories_file,
-    default_stack_config,
-    default_nlu_data,
+    tmp_path: Text,
```

Comment:
Is one directory actually enough or do we need multiple directories?

Reply:
This is the directory from which `tempfile` will derive all of its temporary files/directories, so one is ok.

https://docs.python.org/3/library/tempfile.html#tempfile.tempdir

Reply:
Clever!